Format: add a guide (#1203)
SEKOIA-IO/documentation: _shared_content/develop/guides/formats/parser.md

```json
{
  "message": "LEEF:2|Security|threatmanager|1.0|100|^|src=10.0.0.1^dst=2.1.2.2^spt=1232"
}
```

Below is the parsing pipeline that deserialize the `message` field and set some fields.

```yaml
pipeline:
  - name: parsed_event
    external:
      name: leef.parse-leef
  - name: agent
stages:
  agent:
    actions:
      - set:
          event.id: "{{parsed_event.message.DeviceEventClassID}}"
          source.ip: "{{parsed_event.message.src}}"
          source.port: "{{parsed_event.message.spt}}"
          destination.ip: "{{parsed_event.message.dst}}"
          observer.vendor: "{{parsed_event.message.DeviceVendor}}"
          observer.product: "{{parsed_event.message.DeviceProduct}}"
          observer.version: "{{parsed_event.message.DeviceVersion}}"
```

The following shows the produced event.

```json
{
  "message": "LEEF:2|Security|threatmanager|1.0|100|^|src=10.0.0.1^dst=2.1.2.2^spt=1232"
  "source": {
    "ip": "10.0.0.1",
    "port": 1232
  },
  "destination": {
    "ip": "2.1.2.2"
  },
  "event": {"id": "100"},
  "observer": {
    "vendor": "Security",
    "product": "threatmanager",
    "version": "1.0"
  }
}
```

## Action

> An action is an elementary operations that can create, update and delete fields.

Reviewer: ```suggestion
An action is an elementary operation that can create, update and delete fields.
```

> The execution of an action can be conditionned to a filter.

Reviewer: ```suggestion
The execution of an action can be conditioned by a filter.
```

```

### translate

Sets value of one or more fields according the value of a source field and a dictionary that connect values.

The field in event's final version can be specified with a dotted path (i.e. `field1`, `field1.sub-field1`, …).

An optional fallback value can be defined.
If the value of the source field doesn't match any entry of the mapping dictionary, this fallback value will be used to set the target field.
If no fallback value is defined and the value of the source field doesn't match any entries, the target field will not be created in the final event.

Example:

```yaml
- translate:
  dictionary:
    200: "OK"
    201: "Created"
    204: "No Content"
    400: "Bad Request"
    401: "Unauthorized"
    403: "Forbidden"
    404: "Not Found"
    500: "Internal Server Error"
    501: "Not Implemented"
    502: "Bad Gateway"
    503: "Service Unavailable"
    504: "Gateway Timeout"
  mapping:
    http.response.status_code: http.response.status_message
    api.status_code: api.status_message
  fallback: "Request Processed"
  filter: '{{stage1.log_type == "network"}}'
```

### References

Each stage generates a layer with a set of produced fields in the stage.

> You can refer one of these field from another field, a filter or the input of a next common stage.

Reviewer: ```suggestion
You can refer to one of these fields from another field, a filter or the input of a next common stage.
```

> To refer a field, use a [jinja](https://jinja.palletsprojects.com) placeholder (ie `{{xxxx}}`) with the name of the referred layed (ie, the name of the stage) as the prefix then the path to the field in dot-notation

Reviewer: ```suggestion
To refer to a field, use a [jinja](https://jinja.palletsprojects.com) placeholder (ie `{{xxxx}}`) with the name of the referred layer (ie, the name of the stage) as the prefix then the path to the field in dot-notation
```

> (eg, to refer the value of the field `date` from the `parsed_date` stage, use `{{parsed_date.date}}`).

Reviewer: ```suggestion
(eg, to refer to the value of the field `date` from the `parsed_date` stage, use `{{parsed_date.date}}`).
```
